src/cmd/sanity_test/restore/exchange.go

```diff
@@ -122,7 +123,7 @@ func recursivelyBuildTree(
 	}
 
 	for _, child := range childFolders {
-		if int(ptr.Val(child.GetTotalItemCount())) == 0 {
+		if int(ptr.Val(child.GetTotalItemCount()))+len(childFolders) == 0 {
```

Comment:
I'm okay with these changes, but I'm starting to worry that all of the amendments to "get counts" implies that those funcs are counting the wrong things.  This is the, like, sixth one-off fix to a count like this.  It's beginning to look brittle from a maintenance perspective.  Do we need to centralize these changes in the counter func itself?  Or maybe update the func signature to return more data?

Reply:
Hmm, I understand your concern, but I feel like we are just matching the behavior of export and restore with the restores. I've tried to add comments wherever it might not be obvious. But yeah, I'll keep an eye.